R/RSocrata.R

```diff
@@ -19,6 +19,38 @@ logMsg <- function(s) {
   cat(format(Sys.time(), "%Y-%m-%d %H:%M:%OS3 "), as.character(sys.call(-1))[1], ": ", s, '\n', sep='')
 }
 
+#' Compiles the information to be used in HTTP headers
+#' 
+#' Grabs the headers (RSocrata version, OS, and R version) to be used while
+#' making HTTP requests with Socrata. This enables Socrata's team to track
+#' the usage of RSocrata.
+#' @return a string
+#' @importFrom utils packageVersion
+#' @author Tom Schenk Jr \email{tom.schenk@@cityofchicago.org}
+#' @noRd
+fetch_user_agent <- function() {
+  rSocrataVersion <- packageVersion("RSocrata")
+  operatingSystem <- Sys.info()[["sysname"]]
+  operatingSystemVersion <- paste(Sys.info()[["release"]], Sys.info()[["version"]])
+  rVersion <- paste0(R.version$major, 
+                     ".", 
+                     R.version$minor, 
+                     ifelse( # Checks if version has status, e.g., "rev"
+                       R.version$status == "", 
+                       "", 
+                       paste0("-",R.version$status))
+  )
+  
+  header <- paste0( "RSocrata/",
+                    rSocrataVersion, " (",
+                    operatingSystem, "/",
+                    operatingSystemVersion, "; ",
+                    "R/", rVersion,
+                    ")"
+  )
+  return(header)
```

Comment:
I would put the call to `httr::user_agent` here, but there is no error as it stands.
e.g.  something like this:

```R
result <- httr::user_agent(header)
return(result)
```

Reply:
I'm a little less inclined for this change. If we make the change, the call from `GET` would be:

```r
GET( url, fetch_user_agent() )
```

But, I think I prefer the literal call-out in the current method:

```r
GET( url, user_agent( fetch_user_agent() ) )
```